### ISSUE TITLE
No upload of code (e.g. py-files)

### ISSUE THREAD

**Issue by buhtz** [dear-github/dear-github#151]
Opening an Issue doesn't allow the user to upload code-files (.py, .cpp, ...). Code files are the basic of the most Issuses. No reason to block that.

Posting big code chunks into the message itself is unpolite and doesn't help someone. Allowing code-files as attachments would prevent users to post their code into the message.

**Comment by Kubo2** [comment]
+1 to this

**Comment by koenpunt** [comment]
I think issues shouldn't really be about implementation, but about the source, whereas the source can easily be in the users fork. IMO Uploading code files to issues isn't really going to help anyone

**Comment by buhtz** [comment]
I talk about "minimal example code" to reproduce errors. Code like that has nothing to do in a open repository on GitHub and is sometimes to much to paste into the message itself.

**Comment by koenpunt** [comment]
"Minimal example code" fits perfectly in a comment

**Comment by buhtz** [comment]
"minimal" doesn't mean "few lines". When a bug is complex the code to reproduce it will become complexe, too.

**Comment by koenpunt** [comment]
We obviously have different opinions on this matter.

**Comment by buhtz** [comment]
It is not an opinon on my site it is my experience and expertise as a software developer on different plattforms and languages.

btw: Who are you? In your profile I can't see any connection to the offical GitHub.

**Comment by koenpunt** [comment]
I'm not affiliated to GitHub, but I for example am one of the maintainers of the [Chosen repository](https://github.com/harvesthq/chosen), where issues about implementation are more-or-less daily business. And these are indeed the issues that require more than a few lines to demonstrate their problem.
And with the amount of users of this plugin, having people upload their code files for maintainers to test is a lot of work; having clean repo state, add the faulty file, compile, run, test. 
So [we encourage users](https://github.com/harvesthq/chosen/blob/master/contributing.md#bug-reports) to create a fiddle, or otherwise publicly accessible example of their bug or issue. 
A fiddle functionality in the GitHub issue system on the other hand could be valuable. Maybe just like StackOverflow does.

**Comment by koenpunt** [comment]
> It is not an opinon on my site it is my experience and expertise

@buhtz it still is an opinion, based on your experience and expertise

**Comment by buhtz** [comment]
Not sure what you mean with _fiddle_. :)

I didn't see to much work for the GitHub-devs to allow e.g. py-files. I think there is a filter-file or whitelist with acceptable file-formats. Just add code files (py, cpp, c, cxx, R, tex, ...) to it, too.

**Comment by koenpunt** [comment]
You're right, fiddles (jsfiddle.net) would only be of use for html/css/js, but I'm sure there are services for other languages as well.

**Comment by SimenB** [comment]
Wouldn't using [GIsts](https://gist.github.com/) solve this? You won't be able to run the files, but it allows you to upload multiple files

**Comment by duncanriach** [comment]
For various reasons, I think a GitHub gist is exactly what should be used if there are too many lines for a reasonable comment or if there are multiple files.
